File: src/data/courses/dsa/arrays-and-hashing/array-manipulation/rotate-image/code/cpp/rotate_image_sol1.cpp
class Solution {
public:
    // Extra Matrix: O(n²) space — Copy to new matrix
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();
        vector<vector<int>> copy = matrix;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                matrix[j][n-1-i] = copy[i][j]; // Place at rotated position
    }
};

File: src/data/courses/dsa/arrays-and-hashing/array-manipulation/rotate-image/code/java/RotateImageSol1.java
class Solution {
    // Extra Matrix: O(n²) space
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        int[][] copy = new int[n][n];
        for (int i = 0; i < n; i++) copy[i] = matrix[i].clone();
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++) matrix[j][n-1-i] = copy[i][j];
    }
}